### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by siddharthkp**
This PR adds the checkpoint 1 of comments adding this way the global comments in the sidebar

This will only show when both the sidebar redesign and the comments flags are on

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3600/commits/79cbe9bbb214f608e17b862903c6334ecf6e7d7f is at https://pr3600.build.csb.dev/s/new.

**Comment by christianalfoni** [comment]
Think it looks good! I guess we should deploy to `stream` to test if it only appears locally?

Btw:

```ts
  const { comments } = await effects.fakeGql.queries.allComments({
    sandboxId: sandbox.id,
  });
```

This is in **sandboxChanged**. This should not run unless comments is active or it will delay the loading of initial sandbox for everyone :-)

**Comment by SaraVieira** [comment]
@christianalfoni 

Added the if statment and also a try catch just to make sure we don't break anything

```
  if (COMMENTS_ON) {
    try {
      const { comments } = await effects.fakeGql.queries.allComments({
        sandboxId: sandbox.id,
      });

      state.editor.comments[sandbox.id] = comments.reduce((aggr, comment) => {
        aggr[comment.id] = comment;

        return aggr;
      }, {});
    } catch (e) {
      state.editor.comments[sandbox.id] = {};
    }
  }
```

**Comment by CompuIves** [comment]
If @christianalfoni approves with the changes this is ready to be merged 😄

File: packages/app/src/app/overmind/namespaces/editor/actions.ts [hunk 1]
```diff
@@ -1380,60 +1381,64 @@ export const addComment: AsyncAction<{
   username: string;
 }> = async ({ state, effects }, { sandboxId, comment, username }) => {
   const id = `${comment}-${username}`;
-  const optimisticComment = {
-    id,
-    insertedAt: new Date().toString(),
-    isResolved: false,
-    originalMessage: {
+  if (state.user) {
```
Comment (MichaelDeBoey) [inline packages/app/src/app/overmind/namespaces/editor/actions.ts]: Why don't you just return if `!state.user`?
This makes the rest of the code less indented, which makes the code more readable imo.

```ts
if(!state.user) {
  return;
}

const optimisticComment = {
  // ...
};
// ...
```

File: packages/app/src/app/overmind/namespaces/editor/actions.ts [hunk 2]
```diff
@@ -1380,60 +1381,64 @@ export const addComment: AsyncAction<{
...
+    }
   }
 };
 
 export const deleteComment: AsyncAction<{
   id: string;
 }> = async ({ state, effects }, { id }) => {
-  const sandboxId = state.editor.currentSandbox.id;
-  const deletedComment = state.editor.comments[sandboxId][id];
+  if (state.editor.currentSandbox) {
```
Comment (MichaelDeBoey) [inline packages/app/src/app/overmind/namespaces/editor/actions.ts]: Same

```ts
if (!state.editor.currentSandbox) {
  return;
}

const sandboxId = state.editor.currentSandbox.id;
const deletedComment = state.editor.comments[sandboxId][id];
// ...
```

File: packages/app/src/app/overmind/namespaces/editor/actions.ts [hunk 3]
```diff
@@ -1444,29 +1449,33 @@ export const updateComment: AsyncAction<{
...
   };
 }> = async ({ state, effects }, { id, data }) => {
-  const sandboxId = state.editor.currentSandbox.id;
-  const isResolved = state.editor.comments[sandboxId][id].isResolved;
-  const comment = state.editor.comments[sandboxId][id].originalMessage.content;
-
-  if ('isResolved' in data) {
-    state.editor.comments[sandboxId][id].isResolved = data.isResolved;
-  }
+  if (state.editor.currentSandbox) {
```
Comment (MichaelDeBoey) [inline packages/app/src/app/overmind/namespaces/editor/actions.ts]: Same

```ts
if (!state.editor.currentSandbox) {
  return;
}

const sandboxId = state.editor.currentSandbox.id;
const isResolved = state.editor.comments[sandboxId][id].isResolved;
// ...
```

File: packages/app/src/app/overmind/namespaces/editor/actions.ts (hunk 1 above)
Comment (christianalfoni) [reply packages/app/src/app/overmind/namespaces/editor/actions.ts]: Good point, fixed :)

File: packages/app/src/app/overmind/namespaces/editor/actions.ts (hunk 2 above)
Comment (christianalfoni) [reply packages/app/src/app/overmind/namespaces/editor/actions.ts]: Good point, fixed :)

File: packages/app/src/app/overmind/namespaces/editor/actions.ts (hunk 3 above)
Comment (christianalfoni) [reply packages/app/src/app/overmind/namespaces/editor/actions.ts]: And jup!

**Pull request merged**